tests/TelemetryUtil.ts

```diff
@@ -24,6 +24,9 @@ export const validateTelemetry: (json: string, numPhrases: number, numHypothesis
         expect(hypothesis.length).toEqual(numHypothesis);
     }
 
+    // For review: V2 service does not appear to be sending a PhraseLatencyMs metric, despite sending a 
+    // translation.phrase message
```

Comment:
The service doesn't send the message, the client does.

Reply:
This is happening because the speech.end is received before translation.phrase.
On handling speech.end, the code calls shrinkBuffers(), which updates the last offset seen.
Then the translation.phrase message is handled, and the telemetry checks the phrase offset to make sure it's greater than the last seen offset. Because of the out-of-order reception, the phrase offset is less than the last seen offset, so telemetry does not push a PhraseLatencyMs object to Metrics, assuming the phrase offset is not known.
(code [here](https://github.com/microsoft/cognitive-services-speech-sdk-js/blob/a79b92f5bd774e03113a196402cedbee82f8cf90/src/common.speech/ServiceTelemetryListener.Internal.ts#L68))